test/common/stats/thread_local_store_test.cc

```diff
@@ -508,10 +508,11 @@ TEST_F(StatsThreadLocalStoreTest, ScopeDelete) {
   EXPECT_CALL(main_thread_dispatcher_, post(_));
   EXPECT_CALL(tls_, runOnAllThreads(_, _)).Times(testing::AtLeast(1));
   scope1.reset();
-  EXPECT_EQ(0UL, store_->counters().size());
+  EXPECT_EQ(1UL, store_->counters().size());
```

Comment:
just add a comment here that the counter is gone from all scopes, but is still held by the local variable so it won't be removed from the allocator or store.

Reply:
Done.